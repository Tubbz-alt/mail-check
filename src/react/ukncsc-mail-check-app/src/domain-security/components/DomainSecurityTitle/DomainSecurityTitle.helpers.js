// eslint-disable-next-line
export const getTitleIconProps = (
  error,
  failures,
  warnings,
  inconclusives,
  pending
) => {
  if (error) {
    return { name: 'close', color: 'red' };
  }

  if (failures && !!failures.length) {
    return { name: 'exclamation circle', color: 'red' };
  }

  if (warnings && !!warnings.length) {
    return { name: 'exclamation triangle', color: 'yellow' };
  }

  if (inconclusives && !!inconclusives.length) {
    return { name: 'question circle', color: 'grey' };
  }

  if (pending) {
    return { name: 'time', color: 'blue' };
  }

  return { name: 'check circle', color: 'green' };
};
