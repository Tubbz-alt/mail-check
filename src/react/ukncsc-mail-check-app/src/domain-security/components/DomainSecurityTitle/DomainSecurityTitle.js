import React from 'react';
import PropTypes from 'prop-types';
import { Header, Icon, Loader } from 'semantic-ui-react';
import moment from 'moment';
import { getTitleIconProps } from './DomainSecurityTitle.helpers';

import './DomainSecurityTitle.css';

const DomainSecurityTitle = ({
  title,
  loading,
  error,
  subtitle,
  failures,
  warnings,
  inconclusives,
  pending,
  lastChecked,
}) => (
  <React.Fragment>
    <Header as="h1">
      {loading && <Loader active inline style={{ marginRight: 20 }} />}
      {!loading && (
        <Icon
          {...getTitleIconProps(
            error,
            failures,
            warnings,
            inconclusives,
            pending
          )}
        />
      )}
      <Header.Content className="DomainSecurityTitle--header">
        {title}
        {subtitle && <Header.Subheader>{subtitle}</Header.Subheader>}
      </Header.Content>
    </Header>
    {lastChecked && <p>Last checked {moment(lastChecked).fromNow()}</p>}
  </React.Fragment>
);

DomainSecurityTitle.defaultProps = {
  title: '',
  loading: false,
  subtitle: null,
  failures: [],
  warnings: [],
  inconclusives: [],
  pending: false,
};

DomainSecurityTitle.propTypes = {
  title: PropTypes.string,
  loading: PropTypes.bool,
  subtitle: PropTypes.string,
  failures: PropTypes.arrayOf(PropTypes.string),
  warnings: PropTypes.arrayOf(PropTypes.string),
  inconclusives: PropTypes.arrayOf(PropTypes.string),
  pending: PropTypes.bool,
};

export default DomainSecurityTitle;
