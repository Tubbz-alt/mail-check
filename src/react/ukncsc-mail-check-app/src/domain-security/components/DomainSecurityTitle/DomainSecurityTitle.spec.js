/* eslint-disable import/first */
jest.mock('moment', () => () => ({ fromNow: () => 'one day ago' }));

import React from 'react';
import { render } from 'react-testing-library';
import 'dom-testing-library/extend-expect';
import DomainSecurityTitle from './DomainSecurityTitle';

describe('DomainSecurityTitle', () => {
  let container;

  describe('when failures are present', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityTitle title="a" failures={['err!']} />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should have a red exclamation circle', () =>
      expect(
        container.getElementsByClassName('red exclamation circle icon')
      ).toHaveLength(1));
  });

  describe('when last checked is present', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityTitle title="a" lastChecked="2018-01-01T12:00" />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should display last checked', () =>
      expect(container.getElementsByTagName('p')[0]).toHaveTextContent(
        'Last checked one day ago'
      ));
  });

  describe('when warnings are present without failures', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityTitle title="a" warnings={['warn!']} />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should have a yellow exclamation triangle', () =>
      expect(
        container.getElementsByClassName('yellow exclamation triangle icon')
      ).toHaveLength(1));
  });

  describe('when inconclusives are present', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityTitle title="a" inconclusives={['inconclusive!']} />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should have a grey question circle', () =>
      expect(
        container.getElementsByClassName('grey question circle icon')
      ).toHaveLength(1));
  });

  describe('when no warnings, failures or inconclusives are present', () => {
    beforeEach(() => {
      ({ container } = render(<DomainSecurityTitle title="a" />));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should have a green check circle', () =>
      expect(
        container.getElementsByClassName('green check circle icon')
      ).toHaveLength(1));
  });
});
