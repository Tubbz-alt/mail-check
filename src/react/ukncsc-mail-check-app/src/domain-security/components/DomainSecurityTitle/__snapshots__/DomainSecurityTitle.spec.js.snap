// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainSecurityTitle when failures are present it should match the snapshot 1`] = `
<div>
  <h1
    class="ui header"
  >
    <i
      aria-hidden="true"
      class="red exclamation circle icon"
    />
    <div
      class="content DomainSecurityTitle--header"
    >
      a
    </div>
  </h1>
</div>
`;

exports[`DomainSecurityTitle when inconclusives are present it should match the snapshot 1`] = `
<div>
  <h1
    class="ui header"
  >
    <i
      aria-hidden="true"
      class="grey question circle icon"
    />
    <div
      class="content DomainSecurityTitle--header"
    >
      a
    </div>
  </h1>
</div>
`;

exports[`DomainSecurityTitle when last checked is present it should match the snapshot 1`] = `
<div>
  <h1
    class="ui header"
  >
    <i
      aria-hidden="true"
      class="green check circle icon"
    />
    <div
      class="content DomainSecurityTitle--header"
    >
      a
    </div>
  </h1>
  <p>
    Last checked 
    one day ago
  </p>
</div>
`;

exports[`DomainSecurityTitle when no warnings, failures or inconclusives are present it should match the snapshot 1`] = `
<div>
  <h1
    class="ui header"
  >
    <i
      aria-hidden="true"
      class="green check circle icon"
    />
    <div
      class="content DomainSecurityTitle--header"
    >
      a
    </div>
  </h1>
</div>
`;

exports[`DomainSecurityTitle when warnings are present without failures it should match the snapshot 1`] = `
<div>
  <h1
    class="ui header"
  >
    <i
      aria-hidden="true"
      class="yellow exclamation triangle icon"
    />
    <div
      class="content DomainSecurityTitle--header"
    >
      a
    </div>
  </h1>
</div>
`;
